proxy/proxy.go

```diff
@@ -512,7 +512,7 @@ func mapRequest(ctx *context, requestContext stdlibcontext.Context, removeHopHea
 
 	rr, err := http.NewRequestWithContext(requestContext, r.Method, u.String(), body)
 	if err != nil {
-		return nil, endpoint, err
+		return nil, u.Host, err
```

Comment:
Lets just `return nil, "", err`. It was a bad idea to return `return nil, endpoint, err` since caller never checks endpoint if there is an error.